tests/test_csv_splitter.py

```diff
@@ -0,0 +1,72 @@
+from main import validate_file_length, is_input_valid, get_file_chunks, create_files
+import pytest
+import os
+from mock_data import CHUNKS, CSV_DATA, TITLE
+from typing import Callable
+from utils_types import StringGeneratorType
+
+
+def create_temp_file_fixture(file_extension: str) -> Callable[[], StringGeneratorType]:
+    @pytest.fixture
+    def temp_file() -> StringGeneratorType:
+        file_name = f"test.{file_extension}"
+        with open(file_name, "w") as test_file:
+            test_file.write(CSV_DATA)
+        yield file_name
+        os.remove(file_name)
+
+    return temp_file
+
+
+temp_csv_file = create_temp_file_fixture(file_extension="csv")
+temp_txt_file = create_temp_file_fixture(file_extension="txt")
+
+
+@pytest.fixture
+def create_files_from_chunks():
+    create_files(chunks=CHUNKS, output_file_name="out")
+    yield
+    os.remove("out_1.csv")
+    os.remove("out_2.csv")
+
+
+def test_validate_file_length_success(temp_csv_file):
+    assert validate_file_length(input_file=temp_csv_file, row_limit=3) is True
+
+
+def test_validate_file_length_should_fail_row_limit_too_high(temp_csv_file):
+    assert validate_file_length(input_file=temp_csv_file, row_limit=8) is False
+
+
+def test_is_input_valid_success(temp_csv_file):
+    assert is_input_valid(input_file=temp_csv_file, row_limit=5) is True
+
+
+def test_is_input_valid_should_fail_file_not_exist():
+    assert is_input_valid(input_file="non_existent_file.csv", row_limit=3) is False
+
+
+def test_is_input_valid_should_fail_not_csv(temp_txt_file):
+    assert is_input_valid(input_file=temp_txt_file, row_limit=3) is False
+
+
+def test_get_file_chunks(temp_csv_file):
+    chunks = get_file_chunks(temp_csv_file, 4)
+    assert len(chunks) == 3
+    assert len(chunks[0]) == len(chunks[1]) == 4
+    assert len(chunks[2]) == 2
+    assert all([chunk[0] == TITLE for chunk in chunks])
+    assert chunks[0][3] == "DataA3,DataB3,78,Value3\n"
+    assert chunks[1][2] == "DataA5,DataB5,60,Value5\n"
+    assert chunks[2][1] == "DataA7,DataB7,24,Value7\n"
+
+
+@pytest.mark.usefixtures("create_files_from_chunks")
+def test_create_files():
```

Comment:
I'm not sure what you're doing here. Are you testing your fixture?
If so, that's nice thinking, but we usually don't do it. The purpose of these tests are to test your actual code, not the code for the tests.

Reply:
The thing is, my fixture uses the actual code to create files, so I want to test that the files were actually created (by my real code), but after that I want to remove those files, so thats why its in a fixture to begin with.

Reply:
OK, I didn't understand this, so we have a different problem here :)
We don't want our fixtures to use the actual code we're testing.
The fixtures should only be used for us to setup and teardown the tests.
The reason is that we want isolation between the actual code and the testing code. We don't want our tests to fail, for example, if we did something wrong with the setup or teardown. We want to be able to purely test the code that should be tested.
Think about how you can do it here.